apiserver/facades/agent/uniter/uniter.go

```diff
@@ -2718,6 +2720,43 @@ func (u *UniterAPI) commitHookChangesForOneUnit(unitTag names.UnitTag, changes p
 		modelOps = append(modelOps, modelOp)
 	}
 
+	if len(changes.SecretDeletes) > 0 {
+		result, err := u.SecretsManagerAPI.RemoveSecrets(params.SecretURIArgs{Args: changes.SecretDeletes})
```

Comment:
obviously would be great to get these into the txn

Reply:
yeah. we're getting into 2 phase commit territory if the backend is not juju etc.
i'd like to come back to this when the design has evolved a bit, will leave a todo